### ISSUE TITLE
New-AzResourceGroupDeployment -Confirm continues on error

### ISSUE THREAD

**Issue by Ba4bes** [Azure/arm-template-whatif#45]
**Describe the bug**
When you use New-AzResourceGroupDeployment -Confirm and there is an error in the template, the errormessage is given, but after that the confirm-menu still shows.

**To Reproduce**
1. Create a template with an error (one that would fail Test-AzResourceGroupDeployment
2. Run 
```PowerShell
New-AzResourceGroupDeployment -ResourceGroupName $ResourcegroupName -TemplateFile $TemplateFile -TemplateParameterFile $TemplateParameterFile -Confirm
```

**Expected behavior**
That the error message is displayed and the terminal returns to the prompt.

**Actual behaviour**
The errormessage is displayed and after that the terminal returns the menu:
*Are you sure you want to execute the deployment* etc

**Screenshots**

![image](https://user-images.githubusercontent.com/43912004/79834635-b788d400-83ad-11ea-91f9-1dd53df38870.png)

**Client** [e.g. PowerShell, CLI, API)
```
Name                           Value
----                           -----
PSVersion                      7.0.0
PSEdition                      Core
GitCommitId                    7.0.0
OS                             Microsoft Windows 10.0.18362
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

**Comment by alex-frankel** [comment]
Hey @Ba4bes - we talked amongst ourselves and agree the current behavior is not correct. We will be pushing a fix in the next 2-4 weeks to get this resolved. Thanks for reporting!

**Comment by shenglol** [comment]
The PR containing the fix is out: https://github.com/Azure/azure-powershell/pull/11774.

**Comment by alex-frankel** [comment]
This should be released with version 4.2.0. Please let us know if there are any other issues.
